ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,152 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Upload commands.
+"""
+
+import os
+import uuid
+from contextlib import closing
+from argparse import ArgumentTypeError
+
+from .. _internal import qemu_img
+from .. _internal.units import KiB, MiB
+
+from . import _api
+from . import _options
+from . import _ovirt
+from . import _ui
+
+
+def register(parser):
+    cmd = parser.add_sub_command(
+        "upload-disk",
+        help="Upload disk",
+        func=upload_disk)
+
+    cmd.add_argument(
+        "--sd-name",
+        required=True,
+        help="Name of the storage domain.")
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
+        help="Upload image format (default qcow2).")
+
+    cmd.add_argument(
+        "--sparse",
+        action="store_true",
+        help="Create sparse disk. Cannot be used with raw format on "
+             "block storage.")
+
+    cmd.add_argument(
+        "--id",
+        type=_validate_uuid,
```

Comment:
I was hesitating in where to put parameter validation of the parameters, given some may be command-specific and some global arguments (where is more clear that belong to `_options`). In this case, we should unify both upload and download.

For the time being maybe it could be int, and then we unify them to uuid string, with validation, in a different patch.

Reply:
I have transformed it to a class in `_options`, although probably I change it back to a function. But at least is already in the right place.